pymc3/step_methods/hmc/nuts.py

```diff
@@ -300,25 +302,31 @@ def _single_step(self, left, epsilon):
             error_msg = str(err)
             error = err
         else:
+            # h - H0
             energy_change = right.energy - self.start_energy
             if np.isnan(energy_change):
                 energy_change = np.inf
 
             if np.abs(energy_change) > np.abs(self.max_energy_change):
                 self.max_energy_change = energy_change
             if np.abs(energy_change) < self.Emax:
-                p_accept = min(1, np.exp(-energy_change))
+                # Acceptance statistic 
+		# e^{H(q_0, p_0) - H(q_n, p_n)} max(1, e^{H(q_0, p_0) - H(q_n, p_n)})
+                # Saturated Metropolis accept probability with Boltzmann weight 
+		# if h - H0 < 0
+                log_p_accept = -energy_change + min(0., -energy_change)
```

Comment:
the `-energy_change` here means that the new `log_p_accept` is twice the log of the old `p_accept`. Is that the functional change in the sampler here?

Reply:
oh, i guess this is more accurately calculating `p_accept`, so that the robbins-munro procedure during tuning has an accurate target?

Reply:
Yes, basically it weights the metropolis acceptance ratio `min(1., accept_ratio)` with multinominal weight `energy_diff[k]/sum(energy_diff)`. And since we already have `log(sum(energy_diff))`, we ends up with the implementation like this here.